src/events/components/EventsHeader/index.tsx

```diff
@@ -13,41 +13,42 @@ export interface IProps {
   changeView: (view: EventView) => void;
   accessible: boolean;
   view: EventView;
+  availableViews: number[];
 }
 
-const EventsHeader = ({ toggleAccessible, changeView, accessible, view }: IProps) => (
+const viewIcon = (viewType: EventView) => {
+  switch (viewType) {
+    case EventView.CALENDAR: {
+      return <CalendarViewIcon />;
+    }
+    case EventView.LIST: {
+      return <ListViewIcon />;
+    }
+    case EventView.IMAGE: {
+      return <ImageViewIcon />;
+    }
+  }
+};
+
+const changeViewIconButton = (viewType: number, changeView: (view: EventView) => void, view: EventView) => (

```

Comment:
Functions returning JSX should be FunctionComponents.
This will allow for better optimization for React.

Simply returning JSX from a function call will make the operation happen on every render of the parent component. Making it its' own component will let us tweak that, to for example only render if the props to the component change.

Reply:
So change it to something like this?
```ts
export interface IPropsChangeViewIconButton {
  changeView: (view: EventView) => void;
  viewType: number;
  view: EventView;
}

const ChangeViewIconButton = ({ viewType, changeView, view }: IPropsChangeViewIconButton) => (
  <div
    className={classNames(style.choice, {
      [style.choiceActive]: view === viewType,
    })}
    onClick={() => changeView(viewType)}
    tabIndex={0}
  >
    <ViewIcon viewType={viewType} />
  </div>
);
```

Reply:
Yeah, that looks good :)

Reply:
Could also consider moving the component to a file of its' own.
may be a bit more readable in the long run, but I'll leave that up to you.

In general I think move components should have their own file.

Reply:
I agree, they should be their own files, even though they are basically just a summation of other components, since having differently named `IProps` is kinda cluttered